### PR TITLE
1709 restart paying non customer

### PR DESCRIPTION
**Pull request by jefflembeck**
This addresses #1709 

If somebody stops paying us, all together, this allows for that non-customer to subscribe again via restarting an org.

### CODE REVIEW

File: handlers/customer.js [hunk 1]
```diff
@@ -319,43 +327,110 @@ customer.subscribe = function(request, reply) {
...
-              if (typeof subscription === 'string') {
-                request.logger.info("created subscription: ", planInfo);
+          return Customer(loggedInUser).createSubscription(planInfo);
+        })
+        .tap(function(subscription) {
+          if (typeof subscription === 'string') {
+            request.logger.info("created subscription: ", planInfo);
+          }
+          var dropCache = P.promisify(User.new(request).dropCache);
+          return dropCache(loggedInUser);
```
Comment (jonathanmarvens) [inline handlers/customer.js]: @jefflembeck: This isn't required, but to be safe, if I was writing this, I'd write it like this:

```javascript
var user = User.new(request);
var dropCache = P.promisify(user.dropCache, user);

return dropCache(loggedInUser);
```

File: agents/customer.js [hunk 2]
```diff
@@ -67,8 +70,10 @@ Customer.prototype.getStripeData = function(callback) {
       }
 
       if (resp.statusCode === 404) {
-        err = Error('customer not found: ' + self.name);
-        err.statusCode = resp.statusCode;
+        err = Object.assign(new Error('Customer not found: ' + self.name), {
+          code: 'ENOCUSTOMER',
+          statusCode: 404
```
Comment (jonathanmarvens) [inline agents/customer.js]: @jefflembeck: Why not `statusCode: resp.statusCode`? I feel like that's a little bit cleaner. Obviously no biggie though.

File: agents/customer.js [hunk 3]
```diff
@@ -249,8 +254,11 @@ Customer.prototype.getLicenseForOrg = function(orgName, callback) {
       }
 
       if (resp.statusCode === 404) {
-        err = new Error('Customer not found');
-        err.statusCode = 404;
+        err = Object.assign(new Error('Customer not found'), {
+          code: 'ENOCUSTOMER',
+          statusCode: 404
```
Comment (jonathanmarvens) [inline agents/customer.js]: Same as https://github.com/npm/newww/pull/1776/files#r50441374 ...

File: handlers/customer.js (hunk 1 above)
Comment (jefflembeck) [inline handlers/customer.js]: Word. We have an issue in for promisifying `dropCache`, so I think it'll wait until then.

File: agents/customer.js (hunk 2 above)
Comment (jefflembeck) [inline agents/customer.js]: definitely cleaner, but I think I just wanted it to read super explicitly.

**Comment by aredridel** [comment]
I think this is ready to :sheep:

It'd be nice to have the couple of todos captured as issues, I think you got one of them already.

**Comment by jefflembeck** [comment]
I grabbed the promisifying of dropCache and of delSession and setSession. I _think_ that's everything. Merging into the feature branch.

**Pull request merged**